tests/Serializer/SerializerTest.php

```diff
@@ -96,6 +97,73 @@ public function testNull(bool $serializeAllObjects): void
         $this->assertNull($result);
     }
 
+    public function testRegisteredObjectSerializers(): void
+    {
+        $object = new class() {
+            public function getPurpose(): string
+            {
+                return 'To be tested!';
+            }
+        };
+
+        $objectClass = \get_class($object);
+
+        $serializer = $this->createSerializer(new Options([
+            'class_serializers' => [
+                $objectClass => $customSerializerCallback = static function ($object): array {
+                    return [
+                        'purpose' => $object->getPurpose(),
+                    ];
+                },
+            ],
+        ]));
+
+        $this->assertEquals([
+            'class' => $objectClass,
+            'data' => $customSerializerCallback($object),
```

Comment:
Please define statically the expected result. Calling the same code used in the real scenario does not makes sense because it it's bugged then the test is bugged too